road_safety_scanner.py

```diff
@@ -81,6 +81,20 @@ def __init__(self):
         # Connect the download button to the handleDownload function
         self.ui.downloadResultsButton.clicked.connect(self.handleDownload)
 
+        # Create a QLabel to display the selected AI
+        self.selectedAILabel = QLabel("Current AI: None", self)
+        self.selectedAILabel.setGeometry(10, 40, 200, 30)  
+        self.selectedAILabel.setStyleSheet("font-size: 16px; color: black;") 
+        
+        # Initializes the selected AI
+        self.currentAI = "ChatGPT"
+        self.selectedAILabel.setText(f"Current AI: {self.currentAI}")
+
+    def onAISelected(self, ai_name):
```

Comment:
Please incorporate this code into the selectAI function below- this new function isn't needed

Reply:
<img width="621" alt="Screenshot 2024-09-20 at 12 33 43" src="https://github.com/user-attachments/assets/0990c50e-9e03-41d8-9216-ad263836c22a">
<img width="742" alt="Screenshot 2024-09-20 at 12 34 05" src="https://github.com/user-attachments/assets/392d17
<img width="727" alt="Screenshot 2024-09-20 at 12 34 19" src="https://github.com/user-attachments/assets/82c83e4b-fe74-46dc-aff8-b8d99f4d946c">
90-e800-4da1-84b3-25447076b4c9">
<img width="744" alt="Screenshot 2024-09-20 at 12 34 29" src="https://github.com/user-attachments/assets/c440864f-f645-4868-b05e-a2d5cf7d0e21">

Reply:
Hi Klein,

This looks good. Just a very minor change, if we can decrease the size "Current AI: xxx" to maybe 12px so its as big as the title (Sorry to be annoying).
![image](https://github.com/user-attachments/assets/87c830bb-9a7d-485b-bcce-7a03bb29f556)
 
And just following up on what James mentioned to use Designer instead of manually coding it in. It will save time and effort. I noticed that there is a error - selectedAILabel is not an attribute in the gui.py file. Using Qt Creator/Designer will solve this rather than coding it in yourself. 
![image](https://github.com/user-attachments/assets/0e894b1d-fee9-4c7c-95db-6f16ed05b6ee)

Reply:
have solve it by using qt Creator. If there is any problem please let me know.